libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -61,34 +76,44 @@ public new BotFrameworkAdapter Use(IMiddleware middleware)
 
         public override async Task SendActivity(IBotContext context, params Activity[] activities)
         {
+            AssertBotFrameworkContext (context);
+
+            BotFrameworkBotContext bfContext = context as BotFrameworkBotContext;
+            MicrosoftAppCredentials appCredentials = await GetAppCredentials(bfContext.BotAppId);
```

Comment:
Hmm, For delay activities this is not really required. Though it probably wont really have a perf impact (Dictionary access and all that) but for clarity probably worth moving it inside the if condition. Assert can be left outside though.